src/time_integration/paired_explicit_runge_kutta/methods_PERK3.jl

```diff
@@ -0,0 +1,366 @@
+    for i in 2:(num_stages - 2)
+        c[i] = cS2 * (i - 1) / (num_stages - 3)
+    end
+
+    return c
+end
+
+# Compute the Butcher tableau for a paired explicit Runge-Kutta method order 3
+# using a list of eigenvalues
+function compute_PairedExplicitRK3_butcher_tableau(num_stages, tspan,
+                                                   eig_vals::Vector{ComplexF64};
+                                                   verbose = false, cS2)
+    # Initialize array of c
+    c = compute_c_coeffs(num_stages, cS2)
+
+    # Initialize the array of our solution
+    a_unknown = zeros(num_stages - 2)
+
+    # Special case of e = 3
+    if num_stages == 3
+        a_unknown = [0.25]
+    else
+        # Calculate coefficients of the stability polynomial in monomial form
+        consistency_order = 3
+        dtmax = tspan[2] - tspan[1]
+        dteps = 1.0f-9
+
+        num_eig_vals, eig_vals = filter_eig_vals(eig_vals; verbose)
+
+        monomial_coeffs, _ = bisect_stability_polynomial(consistency_order,
+                                                         num_eig_vals, num_stages,
+                                                         dtmax, dteps,
+                                                         eig_vals; verbose)
+        monomial_coeffs = undo_normalization!(monomial_coeffs, consistency_order,
+                                              num_stages)
+
+        # Solve the nonlinear system of equations from monomial coefficient and
+        # Butcher array abscissae c to find Butcher matrix A
+        # This function is extended in TrixiNLsolveExt.jl
+        a_unknown = solve_a_butcher_coeffs_unknown!(a_unknown, num_stages,
+                                                    monomial_coeffs, cS2, c;
+                                                    verbose)
+    end
+    # Fill A-matrix in P-ERK style
+    a_matrix = zeros(num_stages - 2, 2)
+    a_matrix[:, 1] = c[3:end]
+    a_matrix[:, 1] -= a_unknown
+    a_matrix[:, 2] = a_unknown
+
+    return a_matrix, c
+end
+
+# Compute the Butcher tableau for a paired explicit Runge-Kutta method order 3
+# using provided values of coefficients a in A-matrix of Butcher tableau
+function compute_PairedExplicitRK3_butcher_tableau(num_stages,
+                                                   base_path_a_coeffs::AbstractString;
+                                                   cS2)
+
+    # Initialize array of c
+    c = compute_c_coeffs(num_stages, cS2)
+
+    # - 2 Since First entry of A is always zero (explicit method) and second is given by c_2 (consistency)
+    a_coeffs_max = num_stages - 2
+
+    a_matrix = zeros(a_coeffs_max, 2)
+    a_matrix[:, 1] = c[3:end]
+
+    path_a_coeffs = joinpath(base_path_a_coeffs,
+                             "a_" * string(num_stages) * ".txt")
+
+    @assert isfile(path_a_coeffs) "Couldn't find file"
+    a_coeffs = readdlm(path_a_coeffs, Float64)
+    num_a_coeffs = size(a_coeffs, 1)
+
+    @assert num_a_coeffs == a_coeffs_max
```

Comment:
Hm.. if the intention is that the user doesn't need to pass `num_stages` then we can also simply try to look for any files with the pattern `a_$num_stages.txt` and extract num_stages from the file's name right away.

However, according to my understanding, this would be a problem since maybe if the user has multiple a coefficients file of different `num_stages` they want to try out, they wouldn't be able to specify which one they want to use.

Please correct me if I'm misunderstanding something.

Reply:
Yes, reading `num_stages` from the filename doesn't seem to work. What do you think of the following solution? The user specifies the path to the file *including the file name*. This way you wouldn't need to enforce a specific naming pattern of the files and you can read `num_stages` from the file, such that the user doesn't need to pass it to the function.

Reply:
That's a great idea! We probably should also adapt this approach to one of the constructors of PERK2 as well.

Reply:
Ah, you have this pattern already in the PERK2 integrator. I would advocate adapting the approach also there, but then this would technically be a breaking change.

Reply:
This type of constructor is meant to be for testing purposes since I don't think the case of the user having the coefficients of Butcher tableau (for PERK3) or the monomial coefficients (for PERK2) of the integrator is a very common one. So I think it should be fine to change this according to your suggestion.

Reply:
Yes, generally I agree. There is probably no one out there who used this constructor except for you and Daniel :)
But from a software development perspective it would still be breaking because there _could_ be someone. Let's see what the others say.

Reply:
We could think about changing this particular constructor such that the actual filename is also given. Makes it somewhat more flexible.

Reply:
Not sure though if this is important enough to turn this into a breaking change and thus somewhat slow down the progress of merging this.

Reply:
I'll revert this change and maybe after this PR is merged we can make another PR regarding this.